### DISCUSSION TITLE
I made a VM manager, which supports 86Box firstly.

### DISCUSSION THREAD

**Discussion by ghost** [86Box/86Box#1440] [Show and tell]
Hi everyone!

I have created a new software that allows the users to create and manage their PC emulator based virtual machines in a graphical way, offering similar features as 86Box Manager but it can do more than that. If you want to see the full list of features, you can find them on my [GitHub page](https://github.com/laciba96/WinBox-Reloaded).

The first emulator it supports is the 86Box, but I want to add more PCem forks, and other emulators later.

Cheers!

**Comment by OBattler** [comment]
It look awesome! It just has a few bugs currently, eg. the "None" for devices is in Hungarian even with the language set to English, and I believe some hangs were reported, but other than that, it's awesome.

**Comment by ppgrainbow** [comment]
It looks awesome so far! As far I know, I tried to run your VM manager, but when it tried to check for updates, the emulator of yours hanged. :(

I'm wondering what could be wrong with it...

**Comment by ts-korhonen** [comment]
Some functions (hdd related) don't work because missing midas.dll. You should probably add MidasLib to the project's uses.

**Comment by ghost** [comment]
Thanks to everyone, I hope I was able to fix some of the earlier bugs you mentioned. I fixed some of the translations and added MidasLib to the project, also I changed the update dialog behaviour. The fix is up as a new Pre-Release.

**Reply by ts-korhonen** [reply]
Now creating VM works, but it still doesn't seem to detect when 86Box has started. It stays at "starting..." while the VM has already started and if I press update list, it goes "Stopped" even though it is running.

**Reply by ghost** [reply]
Can you make an issue with this on my GitHub? I need more information to recreate this problem, because I've never met anything like this during my testing process.

The thing I need is what Windows version do you use, and what language, and I need the path of the Working Directory, and the values from the Profile Settings of the VM you mentioned. Also I need the full command line of the new 86Box process, started by WinBox. You can get the command line by Process Explorer or similar tools.

I appreciate if you take time for these, and I hope I can solve this problem too, but it will surely need more follow-up than the previous ones. Thanks in advance!

**Reply by ts-korhonen** [reply]
Sure, I'll make the issue with more info soon.

**Comment by explorerdotexe** [comment]
This project is absolutely awesome, hoping it can be even more awesome in the future.

**Comment by ts-korhonen** [comment]
@laciba96 
There has been some talk with the developers of 86Box about replacing the current 86Box Manager with your Winbox as the "official" VM manager of 86Box. To do this we feel there should be a version that always uses 86Box to not confuse users that download it from our project page. How would you feel about making a fork of it where it's rebranded as 86Box specific thing? With minimal changes of course to not make maintaining too difficult when you add more emulator support to Winbox.

P.S. please feel welcome to join the 86Box discord server where we discuss about development among other things 🙂

**Reply by ghost** [reply]
Thank you for the opportunity, it is a real honor for me.

Since I don't have time recently to fundamentally change the program, I'm going rebrand the main branch of the original project, and if I still want to add more emulators in the future, I'll create a separate fork.

As for the rebranding, on the one hand, I will remove the texts for multiple emulators, and on other hand I may add some kind of migration options from custom VMs and an existing 86Box installation, although I have not worked out this idea in details yet.

I also joined the Discord server, I can be found as laciba96.

**Reply by ts-korhonen** [reply]
> Since I don't have time recently to fundamentally change the program, I'm going rebrand the main branch of the original project, and if I still want to add more emulators in the future, I'll create a separate fork.

This is a good idea. It'll also keep github issues of the manager in one place.

> As for the rebranding, on the one hand, I will remove the texts for multiple emulators, and on other hand I may add some kind of migration options from custom VMs and an existing 86Box installation, although I have not worked out this idea in details yet.

Pretty much UI text changes and default paths seem to be all that's needed as 86Box don't have much brand styling apart from the program icon.

**Reply by ghost** [reply]
I've made a new release with the UI text, and path changes, and I also added an Import VM Wizard, where you can add 86Box VMs from anywhere your PC, assuming that it's working directory is where the config file is. 

After this wizard, the same behavior applies to these freshly imported virtual machines as to the VMs created by WinBox. This feature hopefully helps the migration from any other VM management solution.

The version of this release is v1.14.

**Reply by ts-korhonen** [reply]
Nice! The import function is a great addition for migrating.

We we're thinking the name should be 86Box Manager like the current one, would you be ok with that? It would make it feel more like using a complete package. Obviously we don't want to take any credit away, so about dialog could very well have "86Box Manager - based on Winbox Reloaded" etc.
